src/main/java/com/mastery/java/task/controllers/EmployeeController.java

```diff
@@ -0,0 +1,157 @@
+        model.addAttribute("loglevel", logLevel);
+    }
+    @ModelAttribute("employees")
+    public void getAllEmployees(Model model){List<Employee> employees = employeeService.getAll();
+        model.addAttribute("employees", employees);}
+
+    @Autowired
+    public EmployeeController(EmployeeService employeeService, JmsMessagingTemplate jmsMessagingTemplate, Queue queue) {
+        this.employeeService = employeeService;
+        this.jmsMessagingTemplate = jmsMessagingTemplate;
+        this.queue = queue;
+    }
+
+    @GetMapping(value = "/")
+    public String home() {
+        logCon.trace("Getting all employees from the database");
+        log.info("Go to the main page");
+        logCon.info("Go to the main page");
+        return "home";
+    }
+
+    @GetMapping("/add")
+    public String employeeAdd(Model model) {
+        logCon.trace("Go to the page for adding an employee");
+        model.addAttribute("employee", new Employee());
+        return "employee-add";
+    }
+
+    @PostMapping("/add")
+    public String employeeAdd(@Valid Employee employee, BindingResult bindingResult, Model model) throws DuplicateEmployeeException, InvalidDateException {
+        if (bindingResult.hasErrors()) {
+            log.info("Invalid employee has " + bindingResult.getFieldErrorCount() + " error");
+            log.trace("Invalid employee: " + bindingResult.getAllErrors());
+            logCon.info("Invalid employee has " + bindingResult.getFieldErrorCount() + " error");
+            model.addAttribute("employee", employee);
+            return "employee-add";
+        }
+        log.info("Employee added: " + employee);
+        logCon.info("Employee added: " + employee);
+        employeeService.save(employee);
+        return "redirect:/";
+    }
+
+    @GetMapping("/get/{id}")
+    public String employeeDetails(@PathVariable(value = "id") int id, Model model) {
+        Employee employee = employeeService.getById(id);
+        log.info("Employee with ID " + id + " received: " + employee);
+        logCon.info("Employee with ID " + id + " received: " + employee);
+        model.addAttribute("employee", employee);
+        return "employee-details";
+    }
+
+    @PostMapping("/{id}/delete")
+    public String employeeDelete(@PathVariable(value = "id") int id) throws InterruptedException {
+        this.jmsMessagingTemplate.convertAndSend(this.queue, String.valueOf(id));
+        log.info("Employee with ID " + id + " deleted");
+        logCon.info("Employee with ID " + id + " deleted");
+        Thread.sleep(50);
+        return "redirect:/";
+    }
+
+    @GetMapping("/{id}/edit")
+    public String employeeEdit(@PathVariable(value = "id") int id, Model model) {
+        Employee employee = employeeService.getById(id);
+        logCon.trace("Go to the page for change an employee with id " + id);
+        model.addAttribute("employee", employee);
+        return "employee-edit";
+    }
+
+    @PostMapping("/{id}/edit")
```

Comment:
this endpoint might have a path '/employee/{id}' with specific http operation, as example Put

Reply:
Done